### DISCUSSION TITLE
Region of interest

### DISCUSSION THREAD

**Discussion by davisalex22** [ultralytics/yolov5#9715] [General]
How can I set a region of interest (ROI) on a video and only detect objects that are in that area of ​​the ROI?
What specific lines need to be modified?

**Comment by glenn-jocher** [comment]
@davisalex22 I'd just crop your image before you pass it for inference.

**Comment by spacewalk01** [comment]
As @glenn-jocher suggested, you can crop your image and feed the cropped image to the model. Here is an example:
```
# Load model
model = torch.hub.load('ultralytics/yolov5', 'custom', path="your model path", force_reload=True)  

# Read image
image = cv2.imread("your image path")
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)

# Crop image
crop = image[100:700, 800:1200] 

# Inference
results = model(crop)
df = results.pandas().xyxy[0]
df = df[df.confidence > 0.2]
```

**Comment by spacewalk01** [comment]
Another approach is to feed the entire image to the model and after detection, filter out those boxes within your ROI. I suggest this approach, because cropping image will reduce the resolution of your image feeding to the model.
